diff --git a/caliptra_ss_tb_rr_arb_tree.sv b/caliptra_ss_tb_rr_arb_tree.sv
index 1aa566d..8beb7ed 100644
--- a/caliptra_ss_tb_rr_arb_tree.sv
+++ b/caliptra_ss_tb_rr_arb_tree.sv
@@ -44,7 +44,7 @@
 /// If it is `1'b1` two `lzc`, a masking logic stage and a two input multiplexer are instantiated.
 /// However these are small in respect of the data multiplexers needed, as the width of the `req_i`
 /// signal is usually less as than `DataWidth`.
-module rr_arb_tree #(
+module caliptra_ss_tb_rr_arb_tree #(
   /// Number of inputs to be arbitrated.
   parameter int unsigned NumIn      = 64,
   /// Data width of the payload in bits. Not needed if `DataType` is overwritten.
@@ -81,7 +81,7 @@ module rr_arb_tree #(
   parameter int unsigned IdxWidth   = (NumIn > 32'd1) ? unsigned'($clog2(NumIn)) : 32'd1,
   /// Dependent parameter, do **not** overwrite.
   /// Type for defining the arbitration priority and arbitrated index signal.
-  parameter type         idx_t      = logic [IdxWidth-1:0]
+  parameter type         caliptra_ss_tb_idx_t      = logic [IdxWidth-1:0]
 ) (
   /// Clock, positive edge triggered.
   input  logic                clk_i,
@@ -90,7 +90,7 @@ module rr_arb_tree #(
   /// Clears the arbiter state. Only used if `ExtPrio` is `1'b0` or `LockIn` is `1'b1`.
   input  logic                flush_i,
   /// External round-robin priority. Only used if `ExtPrio` is `1'b1.`
-  input  idx_t                rr_i,
+  input  caliptra_ss_tb_idx_t                rr_i,
   /// Input requests arbitration.
   input  logic    [NumIn-1:0] req_i,
   /* verilator lint_off UNOPTFLAT */
@@ -106,7 +106,7 @@ module rr_arb_tree #(
   /// Output data.
   output DataType             data_o,
   /// Index from which input the data came from.
-  output idx_t                idx_o
+  output caliptra_ss_tb_idx_t                idx_o
 );
 
   `ifndef SYNTHESIS
@@ -131,12 +131,12 @@ module rr_arb_tree #(
     localparam int unsigned NumLevels = unsigned'($clog2(NumIn));
 
     /* verilator lint_off UNOPTFLAT */
-    idx_t    [2**NumLevels-2:0] index_nodes; // used to propagate the indices
+    caliptra_ss_tb_idx_t    [2**NumLevels-2:0] index_nodes; // used to propagate the indices
     DataType [2**NumLevels-2:0] data_nodes;  // used to propagate the data
     logic    [2**NumLevels-2:0] gnt_nodes;   // used to propagate the grant to masters
     logic    [2**NumLevels-2:0] req_nodes;   // used to propagate the requests to slave
     /* lint_off */
-    idx_t                       rr_q;
+    caliptra_ss_tb_idx_t                       rr_q;
     logic [NumIn-1:0]           req_d;
 
     // the final arbitration decision can be taken from the root of the tree
@@ -148,7 +148,7 @@ module rr_arb_tree #(
       assign rr_q       = rr_i;
       assign req_d      = req_i;
     end else begin : gen_int_rr
-      idx_t rr_d;
+      caliptra_ss_tb_idx_t rr_d;
 
       // lock arbiter decision in case we got at least one req and no acknowledge
       if (LockIn) begin : gen_lock
@@ -205,7 +205,7 @@ module rr_arb_tree #(
 
       if (FairArb) begin : gen_fair_arb
         logic [NumIn-1:0] upper_mask,  lower_mask;
-        idx_t             upper_idx,   lower_idx,   next_idx;
+        caliptra_ss_tb_idx_t             upper_idx,   lower_idx,   next_idx;
         logic             upper_empty, lower_empty;
 
         for (genvar i = 0; i < NumIn; i++) begin : gen_mask
@@ -213,7 +213,7 @@ module rr_arb_tree #(
           assign lower_mask[i] = (i <= rr_q) ? req_d[i] : 1'b0;
         end
 
-        lzc #(
+        caliptra_ss_tb_lzc #(
           .WIDTH ( NumIn ),
           .MODE  ( 1'b0  )
         ) i_lzc_upper (
@@ -222,7 +222,7 @@ module rr_arb_tree #(
           .empty_o ( upper_empty )
         );
 
-        lzc #(
+        caliptra_ss_tb_lzc #(
           .WIDTH ( NumIn ),
           .MODE  ( 1'b0  )
         ) i_lzc_lower (
@@ -235,7 +235,7 @@ module rr_arb_tree #(
         assign rr_d     = (gnt_i && req_o) ? next_idx  : rr_q;
 
       end else begin : gen_unfair_arb
-        assign rr_d = (gnt_i && req_o) ? ((rr_q == idx_t'(NumIn-1)) ? '0 : rr_q + 1'b1) : rr_q;
+        assign rr_d = (gnt_i && req_o) ? ((rr_q == caliptra_ss_tb_idx_t'(NumIn-1)) ? '0 : rr_q + 1'b1) : rr_q;
       end
 
       // this holds the highest priority
@@ -272,7 +272,7 @@ module rr_arb_tree #(
             // arbitration: round robin
             assign sel =  ~req_d[l*2] | req_d[l*2+1] & rr_q[NumLevels-1-level];
 
-            assign index_nodes[Idx0] = idx_t'(sel);
+            assign index_nodes[Idx0] = caliptra_ss_tb_idx_t'(sel);
             assign data_nodes[Idx0]  = (sel) ? data_i[l*2+1] : data_i[l*2];
             assign gnt_o[l*2]        = gnt_nodes[Idx0] & (AxiVldRdy | req_d[l*2])   & ~sel;
             assign gnt_o[l*2+1]      = gnt_nodes[Idx0] & (AxiVldRdy | req_d[l*2+1]) & sel;
@@ -287,7 +287,7 @@ module rr_arb_tree #(
           // if index is out of range, fill up with zeros (will get pruned)
           if (unsigned'(l) * 2 > NumIn-1) begin : gen_out_of_range
             assign req_nodes[Idx0]   = 1'b0;
-            assign index_nodes[Idx0] = idx_t'('0);
+            assign index_nodes[Idx0] = caliptra_ss_tb_idx_t'('0);
             assign data_nodes[Idx0]  = DataType'('0);
           end
         //////////////////////////////////////////////////////////////
@@ -299,8 +299,8 @@ module rr_arb_tree #(
           assign sel =  ~req_nodes[Idx1] | req_nodes[Idx1+1] & rr_q[NumLevels-1-level];
 
           assign index_nodes[Idx0] = (sel) ?
-            idx_t'({1'b1, index_nodes[Idx1+1][NumLevels-unsigned'(level)-2:0]}) :
-            idx_t'({1'b0, index_nodes[Idx1][NumLevels-unsigned'(level)-2:0]});
+            caliptra_ss_tb_idx_t'({1'b1, index_nodes[Idx1+1][NumLevels-unsigned'(level)-2:0]}) :
+            caliptra_ss_tb_idx_t'({1'b0, index_nodes[Idx1][NumLevels-unsigned'(level)-2:0]});
 
           assign data_nodes[Idx0]  = (sel) ? data_nodes[Idx1+1] : data_nodes[Idx1];
           assign gnt_nodes[Idx1]   = gnt_nodes[Idx0] & ~sel;
@@ -348,4 +348,4 @@ module rr_arb_tree #(
     `endif
   end
 
-endmodule : rr_arb_tree
+endmodule : caliptra_ss_tb_rr_arb_tree
