--- a/caliptra_ss_tb_axi_demux.sv
+++ b/caliptra_ss_tb_axi_demux.sv
@@ -14,13 +14,12 @@
 // - Thomas Benz <tbenz@iis.ee.ethz.ch>
 // - Andreas Kurth <akurth@iis.ee.ethz.ch>
 
-`include "common_cells/assertions.svh"
-`include "common_cells/registers.svh"
+`include "common_cells/caliptra_ss_tb_common_cells_registers.svh"
 
 `ifdef QUESTA
 // Derive `TARGET_VSIM`, which is used for tool-specific workarounds in this file, from `QUESTA`,
 // which is automatically set in Questa.
-`define TARGET_VSIM
+`define CALIPTRA_SS_TB_TARGET_VSIM
 `endif
 
 /// Demultiplex one AXI4+ATOP slave port to multiple AXI4+ATOP master ports.
@@ -39,14 +38,14 @@
 ///
 /// Beats on the B and R channel are multiplexed from the master ports to the slave port with
 /// a round-robin arbitration tree.
-module axi_demux #(
+module caliptra_ss_tb_axi_demux #(
   parameter int unsigned AxiIdWidth     = 32'd0,
   parameter bit          AtopSupport    = 1'b1,
-  parameter type         aw_chan_t      = logic,
-  parameter type         w_chan_t       = logic,
-  parameter type         b_chan_t       = logic,
-  parameter type         ar_chan_t      = logic,
-  parameter type         r_chan_t       = logic,
+  parameter type         caliptra_ss_tb_aw_chan_t      = logic,
+  parameter type         caliptra_ss_tb_w_chan_t       = logic,
+  parameter type         caliptra_ss_tb_b_chan_t       = logic,
+  parameter type         caliptra_ss_tb_ar_chan_t      = logic,
+  parameter type         caliptra_ss_tb_r_chan_t       = logic,
   parameter type         axi_req_t      = logic,
   parameter type         axi_resp_t     = logic,
   parameter int unsigned NoMstPorts     = 32'd0,
@@ -60,15 +59,15 @@
   parameter bit          SpillR         = 1'b0,
   // Dependent parameters, DO NOT OVERRIDE!
   parameter int unsigned SelectWidth    = (NoMstPorts > 32'd1) ? $clog2(NoMstPorts) : 32'd1,
-  parameter type         select_t       = logic [SelectWidth-1:0]
+  parameter type         caliptra_ss_tb_select_t       = logic [SelectWidth-1:0]
 ) (
   input  logic                          clk_i,
   input  logic                          rst_ni,
   input  logic                          test_i,
   // Slave Port
   input  axi_req_t                      slv_req_i,
-  input  select_t                       slv_aw_select_i,
-  input  select_t                       slv_ar_select_i,
+  input  caliptra_ss_tb_select_t                       slv_aw_select_i,
+  input  caliptra_ss_tb_select_t                       slv_ar_select_i,
   output axi_resp_t                     slv_resp_o,
   // Master Ports
   output axi_req_t    [NoMstPorts-1:0]  mst_reqs_o,
@@ -84,10 +83,10 @@
   logic slv_ar_ready_chan, slv_ar_ready_sel;
   logic slv_ar_valid_chan, slv_ar_valid_sel;
 
-  select_t slv_aw_select, slv_ar_select;
+  caliptra_ss_tb_select_t slv_aw_select, slv_ar_select;
 
-  spill_register #(
-    .T       ( aw_chan_t  ),
+  caliptra_ss_tb_spill_register #(
+    .T       ( caliptra_ss_tb_aw_chan_t  ),
     .Bypass  ( ~SpillAw   )
   ) i_aw_spill_reg (
     .clk_i,
@@ -99,8 +98,8 @@
     .ready_i ( slv_resp_cut.aw_ready ),
     .data_o  ( slv_req_cut.aw        )
   );
-  spill_register #(
-    .T       ( select_t ),
+  caliptra_ss_tb_spill_register #(
+    .T       ( caliptra_ss_tb_select_t ),
     .Bypass  ( ~SpillAw )
   ) i_aw_select_spill_reg (
     .clk_i,
@@ -116,8 +115,8 @@
   assign slv_resp_o.aw_ready  = slv_aw_ready_chan & slv_aw_ready_sel;
   assign slv_req_cut.aw_valid = slv_aw_valid_chan & slv_aw_valid_sel;
 
-  spill_register #(
-    .T       ( w_chan_t  ),
+  caliptra_ss_tb_spill_register #(
+    .T       ( caliptra_ss_tb_w_chan_t  ),
     .Bypass  ( ~SpillW   )
   ) i_w_spill_reg (
     .clk_i,
@@ -129,8 +128,8 @@
     .ready_i ( slv_resp_cut.w_ready ),
     .data_o  ( slv_req_cut.w        )
   );
-  spill_register #(
-    .T       ( ar_chan_t  ),
+  caliptra_ss_tb_spill_register #(
+    .T       ( caliptra_ss_tb_ar_chan_t  ),
     .Bypass  ( ~SpillAr   )
   ) i_ar_spill_reg (
     .clk_i,
@@ -142,8 +141,8 @@
     .ready_i ( slv_resp_cut.ar_ready ),
     .data_o  ( slv_req_cut.ar        )
   );
-  spill_register #(
-    .T       ( select_t ),
+  caliptra_ss_tb_spill_register #(
+    .T       ( caliptra_ss_tb_select_t ),
     .Bypass  ( ~SpillAr )
   ) i_ar_sel_spill_reg (
     .clk_i,
@@ -159,8 +158,8 @@
   assign slv_resp_o.ar_ready  = slv_ar_ready_chan & slv_ar_ready_sel;
   assign slv_req_cut.ar_valid = slv_ar_valid_chan & slv_ar_valid_sel;
 
-  spill_register #(
-    .T       ( b_chan_t ),
+  caliptra_ss_tb_spill_register #(
+    .T       ( caliptra_ss_tb_b_chan_t ),
     .Bypass  ( ~SpillB  )
   ) i_b_spill_reg (
     .clk_i,
@@ -172,8 +171,8 @@
     .ready_i ( slv_req_i.b_ready    ),
     .data_o  ( slv_resp_o.b         )
   );
-  spill_register #(
-    .T       ( r_chan_t ),
+  caliptra_ss_tb_spill_register #(
+    .T       ( caliptra_ss_tb_r_chan_t ),
     .Bypass  ( ~SpillR  )
   ) i_r_spill_reg (
     .clk_i,
@@ -186,7 +185,7 @@
     .data_o  ( slv_resp_o.r         )
   );
 
-  axi_demux_simple #(
+  caliptra_ss_tb_axi_demux_simple #(
     .AxiIdWidth ( AxiIdWidth  ),
     .AtopSupport( AtopSupport ),
     .axi_req_t  ( axi_req_t   ),
@@ -213,7 +212,7 @@
 // interface wrapper
-`include "axi/assign.svh"
+`include "axi/caliptra_ss_tb_axi_assign.svh"
-`include "axi/typedef.svh"
+`include "axi/caliptra_ss_tb_axi_typedef.svh"
-module axi_demux_intf #(
+module caliptra_ss_tb_axi_demux_intf #(
   parameter int unsigned AXI_ID_WIDTH     = 32'd0, // Synopsys DC requires default value for params
   parameter bit          ATOP_SUPPORT     = 1'b1,
   parameter int unsigned AXI_ADDR_WIDTH   = 32'd0,
@@ -230,51 +229,51 @@
   parameter bit          SPILL_R          = 1'b0,
   // Dependent parameters, DO NOT OVERRIDE!
   parameter int unsigned SELECT_WIDTH   = (NO_MST_PORTS > 32'd1) ? $clog2(NO_MST_PORTS) : 32'd1,
-  parameter type         select_t       = logic [SELECT_WIDTH-1:0] // MST port select type
+  parameter type         caliptra_ss_tb_select_t       = logic [SELECT_WIDTH-1:0] // MST port select type
 ) (
   input  logic    clk_i,                 // Clock
   input  logic    rst_ni,                // Asynchronous reset active low
   input  logic    test_i,                // Testmode enable
-  input  select_t slv_aw_select_i,       // has to be stable, when aw_valid
-  input  select_t slv_ar_select_i,       // has to be stable, when ar_valid
+  input  caliptra_ss_tb_select_t slv_aw_select_i,       // has to be stable, when aw_valid
+  input  caliptra_ss_tb_select_t slv_ar_select_i,       // has to be stable, when ar_valid
   AXI_BUS.Slave   slv,                   // slave port
   AXI_BUS.Master  mst [NO_MST_PORTS-1:0] // master ports
 );
 
-  typedef logic [AXI_ID_WIDTH-1:0]       id_t;
-  typedef logic [AXI_ADDR_WIDTH-1:0]   addr_t;
-  typedef logic [AXI_DATA_WIDTH-1:0]   data_t;
-  typedef logic [AXI_DATA_WIDTH/8-1:0] strb_t;
-  typedef logic [AXI_USER_WIDTH-1:0]   user_t;
-  `AXI_TYPEDEF_AW_CHAN_T(aw_chan_t, addr_t, id_t, user_t)
-  `AXI_TYPEDEF_W_CHAN_T(w_chan_t, data_t, strb_t, user_t)
-  `AXI_TYPEDEF_B_CHAN_T(b_chan_t, id_t, user_t)
-  `AXI_TYPEDEF_AR_CHAN_T(ar_chan_t, addr_t, id_t, user_t)
-  `AXI_TYPEDEF_R_CHAN_T(r_chan_t, data_t, id_t, user_t)
-  `AXI_TYPEDEF_REQ_T(axi_req_t, aw_chan_t, w_chan_t, ar_chan_t)
-  `AXI_TYPEDEF_RESP_T(axi_resp_t, b_chan_t, r_chan_t)
+  typedef logic [AXI_ID_WIDTH-1:0]       caliptra_ss_tb_id_t;
+  typedef logic [AXI_ADDR_WIDTH-1:0]   caliptra_ss_tb_addr_t;
+  typedef logic [AXI_DATA_WIDTH-1:0]   caliptra_ss_tb_data_t;
+  typedef logic [AXI_DATA_WIDTH/8-1:0] caliptra_ss_tb_strb_t;
+  typedef logic [AXI_USER_WIDTH-1:0]   caliptra_ss_tb_user_t;
+  `CALIPTRA_SS_TB_AXI_TYPEDEF_AW_CHAN_T(caliptra_ss_tb_aw_chan_t, caliptra_ss_tb_addr_t, caliptra_ss_tb_id_t, caliptra_ss_tb_user_t)
+  `CALIPTRA_SS_TB_AXI_TYPEDEF_W_CHAN_T(caliptra_ss_tb_w_chan_t, caliptra_ss_tb_data_t, caliptra_ss_tb_strb_t, caliptra_ss_tb_user_t)
+  `CALIPTRA_SS_TB_AXI_TYPEDEF_B_CHAN_T(caliptra_ss_tb_b_chan_t, caliptra_ss_tb_id_t, caliptra_ss_tb_user_t)
+  `CALIPTRA_SS_TB_AXI_TYPEDEF_AR_CHAN_T(caliptra_ss_tb_ar_chan_t, caliptra_ss_tb_addr_t, caliptra_ss_tb_id_t, caliptra_ss_tb_user_t)
+  `CALIPTRA_SS_TB_AXI_TYPEDEF_R_CHAN_T(caliptra_ss_tb_r_chan_t, caliptra_ss_tb_data_t, caliptra_ss_tb_id_t, caliptra_ss_tb_user_t)
+  `CALIPTRA_SS_TB_AXI_TYPEDEF_REQ_T(axi_req_t, caliptra_ss_tb_aw_chan_t, caliptra_ss_tb_w_chan_t, caliptra_ss_tb_ar_chan_t)
+  `CALIPTRA_SS_TB_AXI_TYPEDEF_RESP_T(axi_resp_t, caliptra_ss_tb_b_chan_t, caliptra_ss_tb_r_chan_t)
 
   axi_req_t                     slv_req;
   axi_resp_t                    slv_resp;
   axi_req_t  [NO_MST_PORTS-1:0] mst_req;
   axi_resp_t [NO_MST_PORTS-1:0] mst_resp;
 
-  `AXI_ASSIGN_TO_REQ(slv_req, slv)
-  `AXI_ASSIGN_FROM_RESP(slv, slv_resp)
+  `CALIPTRA_SS_TB_AXI_ASSIGN_TO_REQ(slv_req, slv)
+  `CALIPTRA_SS_TB_AXI_ASSIGN_FROM_RESP(slv, slv_resp)
 
   for (genvar i = 0; i < NO_MST_PORTS; i++) begin : gen_assign_mst_ports
-    `AXI_ASSIGN_FROM_REQ(mst[i], mst_req[i])
-    `AXI_ASSIGN_TO_RESP(mst_resp[i], mst[i])
+    `CALIPTRA_SS_TB_AXI_ASSIGN_FROM_REQ(mst[i], mst_req[i])
+    `CALIPTRA_SS_TB_AXI_ASSIGN_TO_RESP(mst_resp[i], mst[i])
   end
 
-  axi_demux #(
+  caliptra_ss_tb_axi_demux #(
     .AxiIdWidth     ( AXI_ID_WIDTH  ), // ID Width
     .AtopSupport    ( ATOP_SUPPORT  ),
-    .aw_chan_t      (  aw_chan_t    ), // AW Channel Type
-    .w_chan_t       (   w_chan_t    ), //  W Channel Type
-    .b_chan_t       (   b_chan_t    ), //  B Channel Type
-    .ar_chan_t      (  ar_chan_t    ), // AR Channel Type
-    .r_chan_t       (   r_chan_t    ), //  R Channel Type
+    .caliptra_ss_tb_aw_chan_t      (  caliptra_ss_tb_aw_chan_t    ), // AW Channel Type
+    .caliptra_ss_tb_w_chan_t       (   caliptra_ss_tb_w_chan_t    ), //  W Channel Type
+    .caliptra_ss_tb_b_chan_t       (   caliptra_ss_tb_b_chan_t    ), //  B Channel Type
+    .caliptra_ss_tb_ar_chan_t      (  caliptra_ss_tb_ar_chan_t    ), // AR Channel Type
+    .caliptra_ss_tb_r_chan_t       (   caliptra_ss_tb_r_chan_t    ), //  R Channel Type
     .axi_req_t      (  axi_req_t    ),
     .axi_resp_t     ( axi_resp_t    ),
     .NoMstPorts     ( NO_MST_PORTS  ),
