diff --git a/caliptra_ss_tb_spill_register.sv b/caliptra_ss_tb_spill_register.sv
index 80ff37f..85b656a 100644
--- a/caliptra_ss_tb_spill_register.sv
+++ b/caliptra_ss_tb_spill_register.sv
@@ -14,7 +14,7 @@
 
 /// Wrapper around the flushable spill register to maintain back-ward
 /// compatibility.
-module spill_register #(
+module caliptra_ss_tb_spill_register #(
   parameter type T      = logic,
   parameter bit  Bypass = 1'b0     // make this spill register transparent
 ) (
@@ -28,7 +28,7 @@ module spill_register #(
   output T     data_o
 );
 
-  spill_register_flushable #(
+  caliptra_ss_tb_spill_register_flushable #(
     .T(T),
     .Bypass(Bypass)
   ) spill_register_flushable_i (
